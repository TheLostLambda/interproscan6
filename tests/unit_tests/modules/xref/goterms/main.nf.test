nextflow_process {

    name "Test Process GOTERMS"
    script "interproscan/modules/xrefs/goterms/main.nf"
    process "GOTERMS"

    stage {
        copy "data/xrefs/goterms.json"
        copy "data/xrefs/goterms.ipr.json"
        copy "interproscan/scripts/xrefs/goterms.py"
        copy "tests/unit_tests/test_inputs/xrefs/match_entries.json"
    }

    test("Should run without failures") {

        when {
            params {
            }
            process {
                """
                input[0] = file("meta/tests/unit_tests/test_inputs/xrefs/match_entries.json")
                input[1] = "../../../meta/data/xrefs/goterms"
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
