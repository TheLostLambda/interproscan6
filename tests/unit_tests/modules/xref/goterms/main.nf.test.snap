{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "goterms_match_entries.json:md5,30f369f96eed362f63ded58c6476d5e6"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-09-06T18:16:40.82924"
    }
}